Sources/Helpers/OAProfileSettingsResetHelper.m

```diff
@@ -0,0 +1,94 @@
+{
+    [OAAppSettings.sharedManager resetAllProfileSettingsForMode:appMode];
+    [OAAppData.defaults resetProfileSettingsForMode:appMode];
+    
+    NSDictionary* appModeDict = [NSDictionary dictionaryWithObject:appMode forKey:kResetingAppModeKey];
+    [[NSNotificationCenter defaultCenter] postNotificationName:kResetWidgetsSettingsNotification object:nil userInfo:appModeDict];
+    
+    if ([OAAppSettings sharedManager].applicationMode == appMode)
+        [[NSNotificationCenter defaultCenter] postNotificationName:kUpdateWidgestsVisibilityNotification object:nil userInfo:nil];
+    
+    if (appMode.isCustomProfile)
+        [self restoreCustomProfileFromBackup:appMode];
+    else
+        [OAMapStyleSettings.sharedInstance resetMapStyleForAppMode:appMode.variantKey];
+}
+
++ (void) restoreCustomProfileFromBackup:(OAApplicationMode *)appMode
+{
+    NSError *err = nil;
+    NSDictionary *initialJson = @{
+        @"type" : @"PROFILE",
+        @"file" : [NSString stringWithFormat:@"profile_%@.json", appMode.stringKey],
+        @"appMode" : appMode.toJson
+    };
+    
+    OASettingsItem *item = [[OAProfileSettingsItem alloc] initWithJson:initialJson error:&err];
+    OASettingsItemJsonReader *jsonReader = [[OASettingsItemJsonReader alloc] initWithItem:item];
+    [OAProfileSettingsResetHelper restoreFromBackup:[NSString stringWithFormat:@"profile_%@", appMode.stringKey] actor:jsonReader];
+    [[[OsmAndApp instance] mapSettingsChangeObservable] notifyEvent];
+}
+
++ (void) applyReadedSettings:(NSDictionary<NSString *, NSString *> *)settings actor:(OASettingsItemJsonReader *)actor
+{
+    NSMutableDictionary<NSString *, NSString *> *rendererSettings = [NSMutableDictionary new];
+    NSMutableDictionary<NSString *, NSString *> *routingSettings = [NSMutableDictionary new];
+
+    [settings enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull key, NSString * _Nonnull obj, BOOL * _Nonnull stop) {
+        if ([key hasPrefix:@"nrenderer_"] || [key isEqualToString:@"displayed_transport_settings"])
+            [rendererSettings setObject:obj forKey:key];
+        else if ([key hasPrefix:@"prouting_"])
+            [routingSettings setObject:obj forKey:key];
+        else
+            [actor.item readPreferenceFromJson:key value:obj];
+    }];
+
+    [actor.item applyRendererPreferences:rendererSettings];
+    [actor.item applyRoutingPreferences:routingSettings];
+
+    [OsmAndApp.instance.data.mapLayerChangeObservable notifyEvent];
+}
+
++ (void) saveToBackup:(NSDictionary<NSString *, NSString *> *)settings withFilename:(NSString *)filename
+{
+    NSFileManager *fileManager = NSFileManager.defaultManager;
+    NSString *backupFolderPath = [[NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject] stringByAppendingPathComponent:@"osfBackup"];
+    
+    BOOL isDir = YES;
+    if (![fileManager fileExistsAtPath:backupFolderPath isDirectory:&isDir])
```

Comment:
Note

Attempting to predicate behavior based on the current state of the file system or a particular file on the file system is not recommended. Doing so can cause odd behavior or race conditions. It’s far better to attempt an operation (such as loading a file or creating a directory), check for errors, and handle those errors gracefully than it is to try to figure out ahead of time whether the operation will succeed.  More: https://developer.apple.com/documentation/foundation/nsfilemanager/1415645-fileexistsatpath?language=objc